### PR TITLE
chore: release pre-built binaries for macOS ARM

### PR DESCRIPTION
**Pull request by suzuki-shunsuke**
*Issue #, if available:*

- #222

*Description of changes:*

Build and release pre-built binaries for macOS ARM.

Asset name: `dynein-macos-arm.tar.gz`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/build_binaries_on_new_releases.yml [hunk 1]
```diff
@@ -24,8 +23,11 @@ jobs:
           os: ubuntu-latest
           target: arm-unknown-linux-gnueabihf
         - name: macos
-          os: macos-latest
+          os: macos-13
```
Comment (suzuki-shunsuke) [inline .github/workflows/build_binaries_on_new_releases.yml:26]: - macos-13: amd64
- macos-14: arm64

Comment (StoneDot) [reply .github/workflows/build_binaries_on_new_releases.yml:26]: I confirmed that macos-13 with arm64 and macos-14 with amd64 require larger runners.
https://github.com/actions/runner-images/tree/main?tab=readme-ov-file#available-images

**Review by StoneDot — COMMENTED** [review]
Thank you for putting together this pull request. It's an excellent find!

The code looks great to me. Nevertheless, I'd like to double-check that this workflow functions as intended. Do you have any test runs for this new workflow?

Feel free to look at https://github.com/awslabs/dynein/pull/220#issuecomment-2089443804 for your reference as a sample execution.

**Comment by suzuki-shunsuke** [comment]
Thank you for your review.
I released v0.2.2 in my fork and tested the workflow.

https://github.com/suzuki-shunsuke/dynein/releases/tag/v0.2.2
https://github.com/suzuki-shunsuke/dynein/actions/runs/8995974629

The workflow succeeded and pre-built binaries were released properly.

```console
$ gh release view -R suzuki-shunsuke/dynein --json assets v0.2.2 -q '.assets[].name'
dynein-linux-arm.tar.gz
dynein-linux-arm.tar.gz.sha256
dynein-linux.tar.gz
dynein-linux.tar.gz.sha256
dynein-macos-arm.tar.gz
dynein-macos-arm.tar.gz.sha256
dynein-macos.tar.gz
dynein-macos.tar.gz.sha256
dynein-windows.zip
dynein-windows.zip.sha256
```

**Pull request merged**